guides/assets/javascripts/guides.js

```diff
@@ -102,5 +102,57 @@
       }, 3000);
       e.clearSelection();
     });
+
+    var anchorLinks = document.querySelectorAll(".anchorlink");
+    var navLinks = document.querySelectorAll("#subCol a");
+    var DESKTOP_THRESHOLD = 1024;
+
+    var removeHighlight = function () {
+      for (var i = 0, n = navLinks.length; i < n; i++) {
+        navLinks[i].classList.remove("active");
+      }
+    }
+
+    var updateHighlight = function (elem) {
+      if (!elem) return;
+      removeHighlight();
+      elem.classList.add("active");
+      if (window.innerWidth > DESKTOP_THRESHOLD) {
+        elem.scrollIntoView({ block: 'center', inline: 'end' });
+      }
+    }
+
+    var resetHighlight = function () {
+      var chapters = document.querySelector(".chapters");
+      chapters?.scroll({ top: 0, behavior: "smooth" });
```

Comment:
Should detect current preference for reduced motion and if `no-preference`, use `smooth` scrolling only. See #51499 for detection.